devito/finite_differences/finite_difference.py

```diff
@@ -247,7 +247,7 @@ def make_derivative(expr, dim, fd_order, deriv_order, side, matvec, x0, symbolic
             # Pure number
             pass
 
-        deriv = IndexDerivative(expr*weights, {dim: indices.free_dim})
+        deriv = DiffDerivative(expr*weights, {dim: indices.free_dim}, deriv_order)
```

Comment:
I think we should drop the `if exand/else`  since can just do this part and return `deriv.evaluate` if `expand=true` since `IndexDerivative` are properly evaluable since a few PR ago

Reply:
I worry it opens a Pandora's box that I wouldn't be very willing to deal with at this stage, but I can try

Reply:
I'm now opening a different PR for this

Reply:
How do u suggest we handle the second part of the `if`:

```
if not expand and indices.expr is not None:
```

IIRC `indices.expr` may be None for some low order derivatives. 
I think we would first have to normalizate behavior at the top (always produce `indices.expr`) then we could do what you suggested

Reply:
Hum right, didn't think that indices.expr would be None I guess we can park it for now